lib/puppet/http/resolver/serverlist.rb

```diff
@@ -0,0 +1,38 @@
+class Puppet::HTTP::Resolver::ServerList < Puppet::HTTP::Resolver
+  def initialize(client, server_list: Puppet[:server_list], default_port: Puppet[:masterport])
+    @client = client
+    @server_list = server_list
+    @default_port = default_port
+  end
+
+  def resolve(session, name, ssl_context: nil, &block)
+    success = false
+    @server_list.each do |server|
+      host = server[0]
+      port = server[1] || @default_port
+      uri = URI("https://#{host}:#{port}/status/v1/simple/master")
+      service = Puppet::HTTP::Service.create_service(@client, name, host, port)
+      if check_connection?(uri, session, service, ssl_context: ssl_context)
```

Comment:
It'd be good to keep the signature of `check_connection?` the same as the method we're overriding in the superclass. You could push the URI creation into the `check_connection?` method:

```ruby
def check_connection(session, service, ...)
  uri = service.with_base_url("/status/v1/simple/master")
  @client.get(uri, ...)
```

Reply:
It looks like these two things might be slightly different?

```
[1] pry(#<Puppet::HTTP::Resolver::ServerList>)> service.with_base_url("/status/v1/simple/master")
=> #<URI::HTTPS https://apple.example.com:8140/puppet-ca/v1/status/v1/simple/master>
[2] pry(#<Puppet::HTTP::Resolver::ServerList>)> uri
=> #<URI::HTTPS https://apple.example.com:8140/status/v1/simple/master>
```

which is what we want?

Reply:
Ah yeah, right. Ideally we could create an instance of the `:puppet` service, and call `puppet.get_status` on it:

```ruby
service = Puppet::HTTP::Service.create_service(@client, :puppet, host, port)
service.get_status
```

But the `puppet` service hasn't been implemented yet. For now, let's just pass the `uri` into the check method.